rest_framework_gis/fields.py

```diff
@@ -12,21 +12,76 @@
 __all__ = ['GeometryField', 'GeometrySerializerMethodField']
 
 
+def recursive_round(v, precision):
+    """
+    Round all numbers within an array or nested arrays
+      v: number or nested array of numbers
+      precision: integer value of number of decimals to keep
+    """
+    if hasattr(v, '__iter__'):
+        return tuple(recursive_round(x, precision) for x in v)
+    return round(v, precision)
+
+
+def _rm_dupes(coordinates, close=True):
+    """
+    Return array with no repeating coordinates.
+      coordinates: array of coordinates
+      close: whether there should always be at least two sets of coordinates, 
+        in which case the first set will be repeated twice
+    """
+    output = []
+    for coord in coordinates:
+        if not output or coord != output[-1]:
+            output.append(coord)
+    if close and len(output) == 1:
+        output.append(output[0])
+    return output
+
+# assert _rm_dupes([(1,1), (1,1), (1,1), (1,1)], False) == [(1, 1)]
+# assert _rm_dupes([(1,1), (1,1), (1,1), (1,1)], True) == [(1, 1), (1, 1)]
+# assert _rm_dupes([(1,0), (1,1), (1,1), (0,1)], True) == _rm_dupes([(1,0), (1,1), (1,1), (0,1)], False) == [(1, 0), (1, 1), (0, 1)]
+
+
+def rm_redundant_points(coordinates, geotype):
```

Comment:
`_remove_redundant_points` would be a better name and more consistent with the rest of naming conventions used in the library.
I'd also make this a static method of `GeometryField`.